src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -172,8 +200,26 @@ private AIState getBurnableMaterial()
         return COOK_COOK_FOOD;
     }
 
-    private AIState retrieve()
+    /**
+     * Retrieve cooked food from furnace and put more fuel in if needed.
+     *
+     * If not going anywhere, transition to START_WORKING.
+     * If we need to walk to building, repeat this state.
+     * If not a furnace target or furnace cooking or furnace is missing both uncooked and cooked food, transition to START_WORKING.
+     * If cooked food in furnace, transfer to inventory.
+     * If no fuel in furnace, and no fuel in inventory, then COOK_RETRIEVE_FUEL.
+     * If no fuel in furnace, and fuel in inventory, transfer fuel to furnace, and delay and transition to START_WORKING
+     * If fuel in furnace, delay and transition to START_WORKING
+     *
+     * @return next AIState
+     */
+    private AIState gatherCookedFoodFromFurnace()
     {
+        // TODO: finding a furnace to walk to should either be in here or in a separate AIState rather than in START_WORKING and other places
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)